packages/firestore/src/remote/remote_store.ts

```diff
@@ -924,3 +895,111 @@ export async function remoteStoreApplyPrimaryState(
     remoteStoreImpl.onlineStateTracker.set(OnlineState.Unknown);
   }
 }
+
+/**
+ * Registers a callback function that is invoked when the network status
+ * changes. Multiple callbacks are supported.
+ */
+function addNetworkStatusChangedHandler(
+  remoteStore: RemoteStore,
+  handler: (enabled: boolean) => Promise<void>
+): void {
+  const remoteStoreImpl = debugCast(remoteStore, RemoteStoreImpl);
+  remoteStoreImpl.networkStatusHandlers.push(handler);
+}
+
+/**
+ * If not yet initialized, registers the WatchStream and its network state
+ * callback with `remoteStoreImpl`. Returns the existing stream if one is
+ * already available.
+ *
+ * PORTING NOTE: On iOS and Android, the WatchStream gets registered on startup.
+ * This is not done on Web to allow it to be tree-shaken.
+ */
+function ensureWatchStream(
```

Comment:
Optional: since this is essentially a lazy-initializing getter, you might want to just call this `watchStream` to cut verbosity in all the call sites.

Reply:
I left this as is since this it would require me to rename all occurrences of `writeStream` and `watchStream` (to avoid name clashes at the callsite), which might make the code even more verbose.